packages/pds/src/db/migrations/20230323T202553064Z-rework-seq.ts

```diff
@@ -0,0 +1,100 @@
+import { Kysely, sql } from 'kysely'
+import { Dialect } from '..'
+
+const repoSeqTable = 'repo_seq'
+const repoOpTable = 'repo_op'
+const repoSeqDidIndex = 'repo_seq_did_index'
+const repoSeqCommitIndex = 'repo_seq_commit_index'
+const repoSeqEventTypeIndex = 'repo_seq_event_type_index'
+const repoSeqSequencedAtIndex = 'repo_seq_sequenced_at_index'
+
+export async function up(db: Kysely<unknown>, dialect: Dialect): Promise<void> {
+  await db.schema.dropIndex(repoSeqCommitIndex).execute()
+  await db.schema.dropIndex(repoSeqDidIndex).execute()
+  await db.schema.dropTable(repoSeqTable).execute()
+  await db.schema.dropTable(repoOpTable).execute()
+
+  let builder = db.schema.createTable(repoSeqTable)
+  if (dialect === 'pg') {
+    builder = builder
+      .addColumn('seq', 'bigserial', (col) => col.primaryKey())
+      .addColumn('invalidatedBy', 'bigint')
```

Comment:
I dig the `invalidatedBy` idea!  I wonder if we should also make it a foreign key.

Reply:
ah yeah I'd be game 👍